Update Ruby on Ice 2019 data

moonglum opened the pull request:
Thanks for the great site and project! I also really like the documentation on how to contribute conference data <3

This PR contains:
* A small commit adding my speaker details
* A few improvements to the Ruby on Ice data:
    * Add the links to website and schedule
    * Add the comments that organize it by date
    * Correct the date and published_at fields
    * Fix Carmen's pronouns
    * Reduce the duplication in the text

One thing left to do is adjust the lightning talks. Can you briefly explain what the `start_cue` and `end_cue` are? Are those timestamps? And if so, in which format? If you tell me, I'm happy to adjust it :+1: 

PS: After getting this merged, I will add [eurucamp 2015](https://2015.eurucamp.org) ([Youtube Playlist](https://www.youtube.com/playlist?list=PLE7tQUdRKcyaywLIEdbRSXYPa0hCvaVm_)) next :smile:

marcoroth commented:
Awesome, thank you so much @moonglum! 🤗 

This looks good already!

> Can you briefly explain what the start_cue and end_cue are? Are those timestamps? And if so, in which format? If you tell me, I'm happy to adjust it 👍

Yeah, they are timestamps/offsets from the start of the video if the video itself contains multiple talks. In that case you just specify the minute and second for the start and end of that talk sections, i.e:

```yaml
     # ...
     start_cue: "01:03"
     end_cue: "05:00"
     # ...
```

Here are the lighting talks from RubyConf 2024 if you want an example:
https://github.com/rubyevents/rubyevents/blob/fdff20568aced82068864ebf6bcaf440069746fe/data/rubyconf/rubyconf-2024/videos.yml#L650-L666

https://www.rubyevents.org/talks/lightning-talks-rubyconf-2024

Let me know if that helps or if you need any guidance! Thank you!

moonglum commented:
@marcoroth Sorry for the radio silence, I was on vacation :blush: Thanks for the feedback <3

* Rebased everything after the YAML change :heavy_check_mark: 
* Added the info about the lightning talks
So from my POV, this is ready now :+1:

moonglum commented:
Can you tell me what the test failure means? Why is the thumbnail missing now? :thinking:

marcoroth commented:
The idea was that if we have the exact timestamps for the lightning talks that we could also show the right frame as the thumbnail for the talk, so that they aren't all the same.

That validation only kicks in when you actually provide timestamps and don't add the thumbnails, which is what happened now.

There are two Rake Tasks to manage thumbnails:

`rails download_missing_meta_talks` to download the video from YouTube using yt-dlp

And then: `rails extract_thumbnails` to extract the frames from the video using ffmpeg.

And them you can commit the files and should be good to go.

You can also verify it locally using: `rails verify_thumbnails`.

moonglum commented:
The tooling on this project is awesome, Marco! And thanks for taking the time to explain everything :bow: 

Locally, I still get two errors/warnings:

```
missing static_metadata for child_talk: carolina-karklis-lighting-talk-ruby-on-ice-2019
missing static_metadata for child_talk: amr-abdelwahab-lighting-talk-ruby-on-ice-2019-2
```

Not entirely sure why though :thinking:

marcoroth commented:
Hey @moonglum, it looks like Carolina Karklis' lightning talk is not in the YAML anymore. Is that on purpose/was it wrong before?

Also looks like Amr's talk has a different ID compared with the one that's currently in the YAML file.

moonglum commented:
* Carolina Karklis: Nope, this talk is not in the recording. I can't remember if they disagreed to the recording being put on YouTube, or if they did not talk.
* Amr: This is because the talk "HW & Machine Hacking with Ruby & Rails" was wrongly attributed to Amr, which gave his actual talk a `-2` postfix.

marcoroth reviewed (approved):
@moonglum okay cool, I think we are good to go then. The warnings are only happening because these talks are in your local database and don't have a counterpart in the YAML files anymore, since they were removed.

So we should be good to go then 👍🏼 

Thanks for working on this!

moonglum commented:
Awesome, thanks a lot, Marco! :bow:
